src/software/logger/log_merger.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include <chrono>
+#include <g3log/logmessage.hpp>
+#include <list>
+#include <string>
+#include <unordered_map>
+
+
+/**
+ * Handles merging repeated log messages into a single message
+ */
+class LogMerger
+{
+   public:
+    /**
+     * Creates a merger to hold repeated log messages
+     */
+    LogMerger();
```

Comment:
we could pass an arg here that lets us not merge logs

Reply:
added